include/fiction/algorithms/simulation/sidb/random_sidb_layout_generator.hpp

```diff
@@ -181,30 +184,8 @@ generate_multiple_random_sidb_layouts(const Lyt&
         const auto random_lyt = generate_random_sidb_layout(lyt_skeleton, params);
 
         // indicates if a found SiDB layout is identical to an already found one.
-        bool identical_layout = false;
-        for (const auto& old_lyt : unique_lyts)
-        {
-            // checks if two layouts have an SiDB at the same position
-            uint64_t identical_cell_counter = 0;
-            old_lyt.foreach_cell(
-                [&identical_cell_counter, random_lyt](const auto& cell_old)
-                {
-                    random_lyt.foreach_cell(
-                        [&identical_cell_counter, &cell_old](const auto& cell_new)
-                        {
-                            if (cell_new == cell_old)
-                            {
-                                identical_cell_counter += 1;
-                            }
-                        });
-                });
-
-            // all cells are identical, so the new layout is a duplicate
-            if (identical_cell_counter == random_lyt.num_cells())
-            {
-                identical_layout = true;
-            }
-        }
+        const auto identical_layout = std::any_of(unique_lyts.cbegin(), unique_lyts.cend(), [&](const auto& old_lyt)
+                                                  { return are_layouts_identical(random_lyt, old_lyt); });
```

Comment:
If you wanted, you could put an execution policy macro on this call. It would speed it up on platforms that support it without you having to change anything else. E.g.,
```suggestion
        const auto identical_layout = std::any_of(FICTION_EXECUTION_POLICY_PAR unique_lyts.cbegin(), unique_lyts.cend(), [&](const auto& old_lyt)
                                                  { return are_layouts_identical(random_lyt, old_lyt); });
```

Reply:
Good idea! Thanks!